kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt

```diff
@@ -0,0 +1,182 @@
+  renderingsAndSnapshots: Flow<RenderingAndSnapshot<R>>,
+  outputs: Flow<O>
+) -> T
+
+/**
+ * Launches the [workflow] in a new coroutine in [scope]. The workflow tree is seeded with
+ * [initialSnapshot] and the first value emitted by [inputs].  Subsequent values emitted from
+ * [inputs] will be used to re-render the workflow.
+ *
+ * This is the primary low-level entry point into the workflow runtime. If you are writing an app,
+ * you should probably be using a higher-level entry point that will also let you define UI bindings
+ * for your renderings.
+ *
+ * ## Initialization
+ *
+ * Before starting the actual workflow runtime, this function will invoke [beforeStart] and pass
+ * it the [Flow]s of renderings, snapshots, and outputs, as well as a [CoroutineScope] that the
+ * runtime will be hosted in. The workflow runtime will not be started until after this function
+ * returns. This is to allow the output flow to start being collected before any outputs can
+ * actually be emitted. Collectors that start _after_ [beforeStart] returns may not receive outputs
+ * that are emitted very quickly. The value returned by [beforeStart] will be returned from this
+ * function after the runtime is launched. The [CoroutineScope] passed to [beforeStart] can be used
+ * to do more than just cancel the runtime – it can also be used to start coroutines that will run
+ * until the workflow is cancelled, namely to collect the rendering and output [Flow]s.
+ *
+ * ## Scoping
+ *
+ * The workflow runtime makes use of
+ * [structured concurrency](https://medium.com/@elizarov/structured-concurrency-722d765aa952).
+ * The runtime is started in a specific [CoroutineScope], which defines the context for the entire
+ * workflow tree – most importantly, the [Job] that governs the runtime's lifetime and exception
+ * reporting path, and the [CoroutineDispatcher][kotlinx.coroutines.CoroutineDispatcher] that
+ * decides what on what threads to run workflow code.
+ *
+ * This function creates a child [Job] in [scope] and uses it as the parent for the workflow
+ * runtime, and as the job passed to the [beforeStart] function. This means that if [beforeStart]
+ * calls `CoroutineScope.cancel`, it will cancel the workflow runtime, but will not cancel the
+ * [scope] passed into this function.
+ *
+ * @param scope The [CoroutineScope] in which to launch the workflow runtime. Any exceptions thrown
+ * in any workflows will be reported to this scope, and cancelling this scope will cancel the
+ * workflow runtime.
+ * @param workflow The root workflow to start.
+ * @param inputs Stream of input values for the root workflow. The first value emitted is passed to
+ * the root workflow's [StatefulWorkflow.initialState], and subsequent emissions are passed as
+ * input updates to the root workflow.
+ * @param initialSnapshot If not null, used to restore the workflow.
+ * @param beforeStart Called exactly once with the flows for renderings/snapshots and outputs.
```

Comment:
Insert after first sentence: Called on a newly created child [Job] of the given [scope], which defines the lifetime of the launched workflow tree. Cancelling that job ends the new workflow session.

Reply:
Reworded a bit, WDYT?

Reply:
I still find "It also gets a sub-scope" confusing since the scope is `this`, not a parameter. That's what got me talking about _receiver_ elsewhere, although I take your point that that's a weird term too -- this ain't Smalltalk. _Called on_ doesn't work for you?

Reply:
I'd be curious what the general opinion is on this. I still think of receivers as pretty much regular parameters, with slightly different syntax rules, but I have no reason to think that's a widely-shared mental model.  "Called on" doesn't seem great to me though, since the receiver doesn't know anything about the method call – it makes sense from the _caller's_ perspective, but not from the _implementer's_.